### PR TITLE
Feature/idcom 1988 did sol support explorer

### PR DESCRIPTION
**Pull request by frank-haomeng-wang**

### CODE REVIEW

File: explorer/src/components/account/SolDidAccountSection.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    if (cluster !== Cluster.Custom) {
+      reportError(err, {
+        address: account.pubkey.toBase58(),
+      });
+    }
+  }
+  return <UnknownAccountCard account={account} />;
+}
+
+function display(value: any): JSX.Element {
```
Comment (rado0x54) [inline explorer/src/components/account/SolDidAccountSection.tsx:41]: Why is the code here different to the one in the instruction one? Ideally for the presentation element of SolData this uses the same code for both components.

File: explorer/src/providers/accounts/index.tsx [hunk 2]
```diff
@@ -146,7 +172,61 @@ export function AccountsProvider({ children }: AccountsProviderProps) {
   );
 }
 
-function parseGatewayToken(result: AccountInfo<Buffer>, pubkey: PublicKey):GatewayTokenProgramData {
+function parseDidSolToken(
+  result: AccountInfo<Buffer>,
+  pubkey: PublicKey,
+  cluster: Cluster
+): DidSolProgramData {
+  const parsedData = SolData.decode<SolData>(result.data);
```
Comment (rado0x54) [inline explorer/src/providers/accounts/index.tsx:180]: Ah ok, this function does the mapping between SolData and DidSolProgramData. It's quite the overhead, but I guess it's needed to have a clean data-model separation.

**Review by rado0x54 — CHANGES_REQUESTED** [review]
Please extract the duplicate display code (and other's that might be related) into one higher class / utility.

**Comment by frank-haomeng-wang** [comment]
I have made the cases more specific in sol did account section and also optimize the logic of the structure.
Screen Shot 2022-06-17 at 5 06 32 PM
Screen Shot 2022-06-17 at 11 33 14 AM

**Pull request closed without merging**